torch/utils/_sympy/functions.py

```diff
@@ -589,8 +617,53 @@ def __new__(cls, *args, **assumptions):
         # base creation
         obj = Expr.__new__(cls, *ordered(args), **assumptions)
         obj._argset = args
+
+        if unique_summation_atoms:
+            obj.unique_summation_atoms = unique_summation_atoms
         return obj
 
+    @classmethod
+    def _should_optimize_max_of_unique_summations(cls, args):
+        """Optimization for patterns of the form
+          max(a+b, c+d)
+          max(max(a+b, c+d), e+f) = max(a+b, c+d, e+f) = k
+          max(k, g+h) .. =  max(a+b, c+d, e+f, g+h)..etc
```

Comment:
It's not really clear what these equations are trying to show. You can explain it with more words which would make it clearer. Something like: "Sometimes, we successfully build larger and larger Max expressions where the arguments ... ... ..."

Reply:
addressed